File: oldPEproblems/problem10.cpp
#include<iostream>
#include<ctime>
using namespace std;

/*
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
Find the sum of all the primes below two million.
*/

inline bool IsPrime(long number) {
	if (number == 3)
		return true;
	else if (number % 3 == 0)
		return false;

	for (long k = 5; k*k <= number; k += 2)
	{
		if (number % k == 0)
		{
			return false;
		}
	}

	return true;
}

int main() 
{
	std::clock_t start;
	double duration;
	start = std::clock();

	long long sum = 2;
	long limit = 2000000;

	for (long check = 3; check < limit; check += 2)
	{
		if (IsPrime(check))
		{
			sum += check;
		}
	}

	cout << "answer is " << sum << endl;
	duration = (std::clock() - start) / (double)CLOCKS_PER_SEC;
	std::cout << "printf: " << duration << endl;
	cin.get();
	return 0;
}

File: oldPEproblems/problem10.py

#The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
#Find the sum of all the primes below two million.
import time

# this function will check if a number is prime
def IsPrime(number):
    if number == 3:
        return True
    elif number % 3 == 0:
        return False
    
    for subNumber in xrange(5, int(number**0.5) + 1, 2):    
        if number % subNumber == 0:
            return False
    '''
    subNumber = 5    
    while subNumber*subNumber <= number:
        if number % subNumber == 0:
            return False
        subNumber += 2
    '''

    return True
# function is over

sum = 2
limit = 2000000

start = time.time()
for numberToCheck in range(3, limit, 2):
    if IsPrime(numberToCheck):
        sum = sum + numberToCheck
    

print "Answer is %s" % sum
print time.time() - start